File: cvsi/cvsi-core/src/main/java/com/winify/cvsi/core/dto/ListDto.java
package com.winify.cvsi.core.dto;

import com.winify.cvsi.core.dto.error.ServerResponseStatus;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

import java.io.Serializable;
import java.util.*;
@ApiModel(value = "ListDto<T>")
public class ListDto<T> extends ServerResponseStatus implements Serializable {
    @ApiModelProperty(name = "list", dataType = "List<T>", required = true)
    private List<T> list;

    public ListDto() {
        list = new ArrayList<T>();
    }

    public List<T> getList() {
        return list;
    }

    public ListDto<T> setList(Set<T> set) {
        this.list.addAll(set);
        return this;
    }

    public ListDto<T> sortBy(Comparator<T> comparator){
        Collections.sort(this.list,comparator);
        return this;
    }
}

File: Mobile/app/CVSI/app/src/main/java/android/cvsi/models/ListDto.kt
package android.cvsi.models

import android.cvsi.models.error.ServerResponseStatus
import java.io.Serializable
import java.util.*

class ListDto<T> : ServerResponseStatus(), Serializable {
    private val list: MutableList<T>

    init {
        list = ArrayList<T>()
    }

    fun getList(): List<T> {
        return list
    }

    fun setList(set: Set<T>): ListDto<T> {
        this.list.addAll(set)
        return this
    }

    fun sortBy(comparator: Comparator<T>): ListDto<T> {
        Collections.sort(this.list, comparator)
        return this
    }
}
